lib/bpk-component-chip/src/BpkChip.js

```diff
@@ -18,37 +18,257 @@
+    ...rest
+  } = props;
+
+  const styles = useBpkDynamicStyleSheet(dynamicStyles);
+  const userStyle = [];
+  const innerStyle = [styles.inner];
+  const textStyle = [styles.text];
+  const iconStyle = [styles.icon];
+
+  if (type === CHIP_TYPES.outline) {
+    userStyle.push({ elevation: 0 });
+    innerStyle.push(styles.innerOutline);
+    textStyle.push(styles.textOutline);
+    iconStyle.push(styles.iconOutline);
+  }
+
+  if (selected) {
+    innerStyle.push(styles.innerSelected);
+    textStyle.push(styles.textSelected);
+    if (type === CHIP_TYPES.outline) {
+      innerStyle.push(styles.innerOutlineSelected);
+    }
+  }
+
+  if (disabled) {
+    innerStyle.push(styles.innerDisabled);
+    textStyle.push(styles.textDisabled);
+    iconStyle.push(styles.iconDisabled);
+    if (type === CHIP_TYPES.outline) {
+      innerStyle.push(styles.innerOutlineDisabled);
+      textStyle.push(styles.textOutlineDisabled);
+      iconStyle.push(styles.iconOutlineDisabled);
+    }
+  }
+
+  const themeAttributes = {
+    ...getThemeAttributes(REQUIRED_THEME_ATTRIBUTES, theme),
+  };
+
+  const shouldApplyTheme = themeAttributes && selected && !disabled;
+  const shouldApplyPrimaryTheme =
+    shouldApplyTheme &&
+    type === CHIP_TYPES.primary &&
+    themeAttributes.chipSelectedBackgroundColor &&
+    themeAttributes.chipSelectedTextColor;
+
+  const shouldApplyOutlineTheme =
+    shouldApplyTheme &&
+    type === CHIP_TYPES.outline &&
+    themeAttributes.chipOutlineSelectedBackgroundColor &&
+    themeAttributes.chipOutlineSelectedTextColor;
+
+  if (shouldApplyPrimaryTheme) {
+    innerStyle.push({
+      backgroundColor: themeAttributes.chipSelectedBackgroundColor,
+    });
+    textStyle.push({
+      color: themeAttributes.chipSelectedTextColor,
+    });
+  }
+  if (shouldApplyOutlineTheme) {
+    innerStyle.push({
+      backgroundColor: themeAttributes.chipOutlineSelectedBackgroundColor,
+    });
+    textStyle.push({
+      color: themeAttributes.chipOutlineSelectedTextColor,
+    });
+  }
+
+  if (style) {
+    userStyle.push(style);
+  }
+
+  if (innerChipStyle) {
+    innerStyle.push(innerChipStyle);
+  }
+
+  let adjustedLeadingAccessoryView;
+  const leadingAccessoryViewItemStyle = textStyle.slice(0);
+  if (leadingAccessoryView) {
+    textStyle.push(styles.textWithLeadingAccessoryView);
+    adjustedLeadingAccessoryView = React.cloneElement(leadingAccessoryView, {
+      style: leadingAccessoryViewItemStyle,
+    });
+  }
+
+  let adjustedTrailingAccessoryView;
+  const trailingAccessoryViewItemStyle = textStyle.slice(0);
+  if (trailingAccessoryView) {
+    textStyle.push(styles.textWithTrailingAccessoryView);
+    adjustedTrailingAccessoryView = React.cloneElement(trailingAccessoryView, {
+      style: trailingAccessoryViewItemStyle,
+    });
+  }
 
-  return <BpkChipWrapper {...rest} dismissible={false} />;
+  return (
+    <BpkChipInner
+      accessibilityLabel={accessibilityLabel}
+      disabled={disabled}
```

Comment:
This doesn't do anything.

BpkChipInner passes the `disabled` prop directly to `BpkTouchableOverlay` which passes it to `TouchableWithoutFeedback`. Instead, it should be applied like this:
```
<TouchableWithoutFeedback
...
accessibilityState={{
  diabled: true,
}}
```

Maybe for ease `disabled` should be a prop of `BpkTouchableOverlay`

Reply:
Ah good spot, will change. Is the PR otherwise ok? As this isn't really part of this PR, it's only showing up in the diff because of the file rename